File: app/src/main/java/com/nizar/specialgirl/java/MainActivity.java
package com.nizar.specialgirl.java;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.view.View;
import android.support.design.widget.NavigationView;
import android.support.v4.view.GravityCompat;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;

import com.nizar.specialgirl.R;

public class MainActivity extends AppCompatActivity
        implements NavigationView.OnNavigationItemSelectedListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Snackbar.make(view, "Yaela, Dipelajarin Tuh Kodingannya", Snackbar.LENGTH_LONG)
                        .setAction("Iya Bang Poncoe", null).show();
                //startActivity(new Intent(MainActivity.this, TidakTersedia.class));
            }
        });

        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
                this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
        drawer.setDrawerListener(toggle);
        toggle.syncState();

        NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
        navigationView.setNavigationItemSelectedListener(this);
    }

    @Override
    public void onBackPressed() {
        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
        if (drawer.isDrawerOpen(GravityCompat.START)) {
            drawer.closeDrawer(GravityCompat.START);
        } else {
            super.onBackPressed();
        }
    }

    //RAISED WHEN NAV VIEW ITEM IS SELECTED
    @SuppressWarnings("StatementWithEmptyBody")
    @Override
    public boolean onNavigationItemSelected(MenuItem item) {
        // Handle navigation view item clicks here.
        int id = item.getItemId();

        if (id == R.id.nav_home) {
            startActivity(new Intent(this, TidakTersedia.class));

        } else if (id == R.id.nav_artist) {
            Intent app = new Intent(Intent.ACTION_VIEW, Uri.parse("https://play.google.com/store/apps/developer?id=LuckyTrue+Development"));
            startActivity(app);

        } else if (id == R.id.nav_aboutus) {
            startActivity(new Intent(this, ContohCardView.class));

        } else if (id == R.id.nav_syarat) {
            startActivity(new Intent(this, ContohListView.class));

        } else if (id == R.id.nav_pertanyaan) {
            startActivity(new Intent(this, TidakTersedia.class));

        } else if (id == R.id.nav_login) {
            startActivity(new Intent(this, TidakTersedia.class));

        } else if (id == R.id.nav_daftar) {
            startActivity(new Intent(this, TidakTersedia.class));

        }

        //REFERENCE AND CLOSE DRAWER LAYOUT
        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
        drawer.closeDrawer(GravityCompat.START);
        return true;
    }
}

File: app/src/main/java/com/nizar/specialgirl/kotlin/MainActivity.kt
package com.nizar.specialgirl.kotlin

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.support.design.widget.FloatingActionButton
import android.support.design.widget.NavigationView
import android.support.design.widget.Snackbar
import android.support.v4.view.GravityCompat
import android.support.v4.widget.DrawerLayout
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.MenuItem
import android.view.View

import com.nizar.specialgirl.R

class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val toolbar = findViewById<View>(R.id.toolbar) as Toolbar
        setSupportActionBar(toolbar)

        val fab = findViewById<View>(R.id.fab) as FloatingActionButton
        fab.setOnClickListener { view ->
            Snackbar.make(view, "Yaela, Dipelajarin Tuh Kodingannya", Snackbar.LENGTH_LONG)
                    .setAction("Iya Bang Poncoe", null).show()
            //startActivity(new Intent(MainActivity.this, TidakTersedia.class));
        }

        val drawer = findViewById<View>(R.id.drawer_layout) as DrawerLayout
        val toggle = ActionBarDrawerToggle(
                this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close)
        drawer.setDrawerListener(toggle)
        toggle.syncState()

        val navigationView = findViewById<View>(R.id.nav_view) as NavigationView
        navigationView.setNavigationItemSelectedListener(this)
    }

    override fun onBackPressed() {
        val drawer = findViewById<View>(R.id.drawer_layout) as DrawerLayout
        if (drawer.isDrawerOpen(GravityCompat.START)) {
            drawer.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }

    //RAISED WHEN NAV VIEW ITEM IS SELECTED
    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        // Handle navigation view item clicks here.
        val id = item.itemId

        if (id == R.id.nav_home) {
            startActivity(Intent(this, TidakTersedia::class.java))

        } else if (id == R.id.nav_artist) {
            val app = Intent(Intent.ACTION_VIEW, Uri.parse("https://play.google.com/store/apps/developer?id=LuckyTrue+Development"))
            startActivity(app)

        } else if (id == R.id.nav_aboutus) {
            startActivity(Intent(this, ContohCardView::class.java))

        } else if (id == R.id.nav_syarat) {
            startActivity(Intent(this, ContohListView::class.java))

        } else if (id == R.id.nav_pertanyaan) {
            startActivity(Intent(this, TidakTersedia::class.java))

        } else if (id == R.id.nav_login) {
            startActivity(Intent(this, TidakTersedia::class.java))

        } else if (id == R.id.nav_daftar) {
            startActivity(Intent(this, TidakTersedia::class.java))

        }

        //REFERENCE AND CLOSE DRAWER LAYOUT
        val drawer = findViewById<View>(R.id.drawer_layout) as DrawerLayout
        drawer.closeDrawer(GravityCompat.START)
        return true
    }
}
